ci: bsim tests: Fix for missing twister*.xml

aescolar opened the pull request:
Do not fail during results merge if
twister did not run (or produce an xml).
Depending on what has changed in a PR
and therefore what tests are run this can happen.

aescolar committed:
ci: bsim tests: Fix for missing result xml files

Do not fail during results merge if
twister or the bsim workflows did not run
(or produce an xml).
Depending on what has changed in a PR
and therefore what tests are run this can happen.

Signed-off-by: Alberto Escolar Piedras <alberto.escolar.piedras@nordicsemi.no>

kartben reviewed (dismissed):
lgtm, thanks!

not sure what happened with your super aggressive line wrapping in the commit message tho :)

theob-pro commented:
Added DNM until, the fix is not correct yet. Bash will expand the glob and the file will still not exist.

theob-pro commented on .github/workflows/bsim-tests.yaml:
```diff
@@ -163,8 +163,12 @@ jobs:
           pip3 install junitparser junit2html
-          junitparser merge ./bsim_*/*bsim_results.*.xml ./twister-out/twister.xml junit.xml
-          junit2html junit.xml junit.html
+          xml_files=`find  twister-out bsim_out -name "twister.xml" -or -name "*bsim_results*xml" 2> /dev/null`
```
It's very unlikely to create an error but it's maybe safer to add `.xml` and not just `xml` for the bsim results.

```suggestion
          xml_files=`find  twister-out bsim_out -name "twister.xml" -or -name "*bsim_results*.xml" 2> /dev/null`
```

aescolar replied:
Thanks changed

aescolar committed:
tests/bsim/run_parallel.sh: Do not attempt to run parallel with no tests

If there is no tests in the test list, do not attempt to run
parallel.
This avoids parallel trying to attempt to read the input from
the terminal in that case.

Signed-off-by: Alberto Escolar Piedras <alberto.escolar.piedras@nordicsemi.no>

kartben commented:
@aescolar `junitparser` supports a `--glob` option, I wonder if the below might just work?

```
junitparser merge --glob "./bsim_*/*bsim_results.*.xml" "./twister-out/twister*.xml" junit.xml
```

aescolar commented:
> @aescolar `junitparser` supports a `--glob` option, I wonder if the below might just work?

It does, and it is indeed nicer. Thanks for the input @kartben

kartben commented:
> > @aescolar `junitparser` supports a `--glob` option, I wonder if the below might just work?
> 
> It does, and it is indeed nicer. Thanks for the input @kartben

Cool!
